tool/tsh/tsh.go

```diff
@@ -3165,11 +3166,8 @@ func setClientWebProxyAddr(cf *CLIConf, c *client.Config) error {
 
 			proxyAddress, err = pickDefaultAddr(
 				timeout, cf.InsecureSkipVerify, parsedAddrs.Host, defaultWebProxyPorts)
-
-			// On error, fall back to the legacy behavior
 			if err != nil {
-				log.WithError(err).Debug("Proxy port resolution failed, falling back to legacy default.")
```

Comment:
My change removes this fallback. I'm not too sure what this fallback was effective. It ended up retrying `:3080` which is already tried by `pickDefaultAddr`, and masking the underlying errors that had occurred in `pickDefaultAddr`. If I'm missing some additional functionality/coverage this was providing, let me know and I can add it back in with a `trace.NewAggregate`.

Reply:
One corner case I see is when things time out in 2 seconds, it doesn't mean it's not reachable.